scripts/cards.lua

```diff
@@ -158,3 +158,30 @@ function takeCardFrom(deckOrCard)
     end
     return nil
 end
+
+function takeCardByNameFrom(deckOrCard, name)
+    if deckOrCard ~= nil then
+        if deckOrCard.tag == "Card" and deckOrCard.name == name then
+            return deckOrCard
+        elseif deckOrCard.tag == "Deck" then
+            for _, value in ipairs(deckOrCard.getObjects()) do
+                if value.name == name then
```

Comment:
@gudyfr Do you know the reason why `value.getName()` here breaks?

Reply:
Yes, getObjects() does not return the actual objects, but a table of tables with a few fields per item in the container.